### PR TITLE
Add test for basic controller device function

### PR DESCRIPTION
**Pull request by Hao-Liu**
1) Define the VM with specified controller device and check result
   meets expectation
2) Start the guest and check if start result meets expectation
3) Test the function of started controller device
4) Shutdown the VM and clean up environment

This PR depends on https://github.com/autotest/virt-test/pull/2026

Signed-off-by: Hao Liu <hliu@redhat.com>

### CODE REVIEW

**Comment by waynesun09** [comment]
@Hao-Liu as I try to run your case with 
```
./run -t libvirt --test controller.functional
```
in test log:

```
13:16:31 DEBUG| Found subtest module /home/wayne/work/virt-test/test-providers.d/downloads/io-github-autotest-libvirt/libvirt/tests/src/daemon/functional.py

```

so there is a problem parsing the test name.

File: libvirt/tests/src/controller/functional.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    vm = env.get_vm(vm_name)
+    vm_xml = VMXML.new_from_inactive_dumpxml(vm_name)
+    vm_xml_backup = vm_xml.copy()
+    try:
+        vm_xml.remove_all_device_by_type('controller')
+        remove_all_addresses(vm_xml)
+        remove_usb_devices(vm_xml)
+        setup_controller_xml()
+        setup_os_xml()
+        logging.error(vm_xml)
```
Comment (waynesun09) [inline libvirt/tests/src/controller/functional.py]: why log level as error here? and also in functions upper.

**Comment by Hao-Liu** [comment]
@waynesun09, This looks like the same issue @will-Do mentioned in https://github.com/autotest/tp-libvirt/pull/438. I'll rename these modules to solve this. Thanks.

File: libvirt/tests/src/controller/functional.py (hunk 1 above)
Comment (Hao-Liu) [inline libvirt/tests/src/controller/functional.py]: That was a debugging line I didn't remove. Will fix it. Thanks.

**Comment by Hao-Liu** [comment]
This PR depends on https://github.com/autotest/virt-test/pull/2051

**Comment by waynesun09** [comment]
the depend PRs merged, Test is good on fedora 22, ACK and merging this.

**Pull request merged**